x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/azureopenai/embeddings/AzureOpenAiEmbeddingsServiceSettings.java

```diff
@@ -0,0 +1,339 @@
+import org.elasticsearch.xpack.inference.services.ServiceUtils;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.elasticsearch.xpack.inference.services.ServiceFields.DIMENSIONS;
+import static org.elasticsearch.xpack.inference.services.ServiceFields.MAX_INPUT_TOKENS;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.extractOptionalBoolean;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.extractOptionalString;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.extractRequiredString;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.removeAsType;
+import static org.elasticsearch.xpack.inference.services.azureopenai.AzureOpenAiServiceFields.API_VERSION;
+import static org.elasticsearch.xpack.inference.services.azureopenai.AzureOpenAiServiceFields.DEPLOYMENT_ID;
+import static org.elasticsearch.xpack.inference.services.azureopenai.AzureOpenAiServiceFields.ENCODING_FORMAT;
+import static org.elasticsearch.xpack.inference.services.azureopenai.AzureOpenAiServiceFields.RESOURCE_NAME;
+
+/**
+ * Defines the service settings for interacting with OpenAI's text embedding models.
+ */
+public class AzureOpenAiEmbeddingsServiceSettings implements ServiceSettings {
+
+    public static final String NAME = "azure_openai_service_settings";
+
+    static final String DIMENSIONS_SET_BY_USER = "dimensions_set_by_user";
+    static final String ENCODING_FORMAT_SET_BY_USER = "encoding_format_set_by_user";
+
+    public static AzureOpenAiEmbeddingsServiceSettings fromMap(Map<String, Object> map, ConfigurationParseContext context) {
+        return switch (context) {
+            case REQUEST -> fromRequestMap(map);
+            case PERSISTENT -> fromPersistentMap(map);
+        };
+    }
+
+    private static AzureOpenAiEmbeddingsServiceSettings fromPersistentMap(Map<String, Object> map) {
+        ValidationException validationException = new ValidationException();
+
+        var settings = fromMap(map, validationException);
+
+        Boolean dimensionsSetByUser = removeAsType(map, DIMENSIONS_SET_BY_USER, Boolean.class);
+        if (dimensionsSetByUser == null && map.containsKey(DIMENSIONS)) {
+            validationException.addValidationError(
+                ServiceUtils.missingSettingErrorMsg(DIMENSIONS_SET_BY_USER, ModelConfigurations.SERVICE_SETTINGS)
+            );
+        }
+
+        Boolean encodingFormatSetByUser = removeAsType(map, ENCODING_FORMAT_SET_BY_USER, Boolean.class);
+        if (encodingFormatSetByUser == null && map.containsKey(ENCODING_FORMAT)) {
+            validationException.addValidationError(
+                ServiceUtils.missingSettingErrorMsg(ENCODING_FORMAT_SET_BY_USER, ModelConfigurations.SERVICE_SETTINGS)
+            );
+        }
+
+        if (validationException.validationErrors().isEmpty() == false) {
+            throw validationException;
```

Comment:
In this context the code is parsing a previously persisted config that is being read back from an index. We should assume that when the config was persisted it passed all the validation checks and read the config without throwing a `validationException`. This is very important if the validation changes in a later version and some previously persisted config that was valid at the time cannot be read back due to validation changes.

When a new field is added to the model config and the code needs to handle upgrading an older version of the config a sensible default should be used. If we have validation here it could make some old configs unreadable in the future. 

`OpenAiEmbeddingsServiceSettings` follows the same pattern but I think this is a mistake.

Reply:
Ah - got it. Thanks for the explanation!